tests/integration/acp.go

```diff
@@ -357,134 +247,92 @@ type DeleteDocActorRelationship struct {
 	ExpectedError string
 }
 
-func deleteDocActorRelationshipACP(
-	s *state,
-	action DeleteDocActorRelationship,
-) {
-	if action.NodeID.HasValue() {
-		nodeID := action.NodeID.Value()
-		collections := s.collections[nodeID]
+func deleteDocActorRelationshipACP(s *state, action DeleteDocActorRelationship) {
+	processNode := func(nodeID int) {
 		node := s.nodes[nodeID]
 
-		var collectionName string
-		if action.CollectionID == -1 {
-			collectionName = ""
-		} else {
-			collection := collections[action.CollectionID]
-			if !collection.Description().Name.HasValue() {
-				require.Fail(s.t, "Expected non-empty collection name, but it was empty.", s.testCase.Description)
-			}
-			collectionName = collection.Description().Name.Value()
-		}
+		collectionName, docID := getCollectionAndDocInfo(s, action.CollectionID, action.DocID, nodeID)
+		requestorIdentity := getRequestorIdentity(s, action.RequestorIdentity, nodeID)
 
-		var docID string
-		if action.DocID == -1 || action.CollectionID == -1 {
-			docID = ""
-		} else {
-			docID = s.docIDs[action.CollectionID][action.DocID].String()
-		}
-
-		var targetIdentity string
-		if action.TargetIdentity == -1 {
-			targetIdentity = ""
-		} else {
-			optionalTargetIdentity := getIdentity(s, nodeID, immutable.Some(action.TargetIdentity))
-			if !optionalTargetIdentity.HasValue() {
-				require.Fail(s.t, "Expected non-empty target identity, but it was empty.", s.testCase.Description)
-			}
-			targetIdentity = optionalTargetIdentity.Value().DID
-		}
-
-		var requestorIdentity immutable.Option[acpIdentity.Identity]
-		if action.RequestorIdentity == -1 {
-			requestorIdentity = acpIdentity.None
-		} else {
-			requestorIdentity = getIdentity(s, nodeID, immutable.Some(action.RequestorIdentity))
-			if !requestorIdentity.HasValue() {
-				require.Fail(s.t, "Expected non-empty requestor identity, but it was empty.", s.testCase.Description)
-			}
-		}
-		ctx := db.SetContextIdentity(s.ctx, requestorIdentity)
-
-		deleteDocActorRelationshipResult, err := node.DeleteDocActorRelationship(
-			ctx,
+		result, err := node.DeleteDocActorRelationship(
+			db.SetContextIdentity(s.ctx, requestorIdentity),
 			collectionName,
 			docID,
 			action.Relation,
-			targetIdentity,
+			getTargetIdentity(s, action.TargetIdentity, nodeID),
 		)
 
-		expectedErrorRaised := AssertError(s.t, s.testCase.Description, err, action.ExpectedError)
-		assertExpectedErrorRaised(s.t, s.testCase.Description, action.ExpectedError, expectedErrorRaised)
+		assertACPResult(s, action.ExpectedError, err, action.ExpectedRecordFound, result.RecordFound, "record found")
+	}
 
-		if !expectedErrorRaised {
-			require.Equal(s.t, action.ExpectedError, "")
-			require.Equal(s.t, action.ExpectedRecordFound, deleteDocActorRelationshipResult.RecordFound)
-		}
+	executeACPAction(action.NodeID, processNode, s)
+}
+
+func executeACPAction(nodeID immutable.Option[int], processNode func(nodeID int), s *state) {
```

Comment:
question: why did this helper and utility function needed to be introduced?

Reply:
so that we don't have somewhat sensitive logic of of running acp-related actions in 2 different places

Reply:
Similar to my other comment, I don't like this over abstraction and jumping around into helpers only for acp actions.

Reply:
call them "helpers" or not, they fix code duplication problem. And I though this is something that should be appreciated. Otherwise I'm absolutely confused especially when seeing tasks like this #3069

Reply:
I am not sure why you are confused about #3069 in this PR. I do agree it can be more updated with more specific info as I found the #3076 bug after that.